notebooks/03_spatial_data_processing.md

```diff
@@ -333,34 +276,705 @@ plt.show()
 
---- 
 
-## Vignette: House Prices
+
+## Vignette: Airbnb Prices in San Diego
+
+Our second vignette changes the spatial scale to examine the distribution of
+Airbnb listings in San Diego California. We begin by reading the listings, 
+ which are stored in a comma separated file (csv):
+
 
 ```python
 df = pd.read_csv('../data/sandiego/listings.csv')
 len(df)
+
+```
+
+Examination of the dataframe reveals that the location information is encoded
+in the columns `longitude` and `latitude`.  We will use these columns together
+with the `Point` class from **shapely** to create a geodataframe:
+
+```python
+from shapely.geometry import Point
+```
+
+```python jupyter={"outputs_hidden": false}
+geometry = [Point(xy) for xy in zip(df['longitude'], df['latitude'])]
+```
+
+```python
+import geopandas as gpd
+```
+
+```python
+gdf = gpd.GeoDataFrame(df)
+```
+
+```python
+gdf['geometry'] = geometry
+```
+
+```python
+gdf.plot()
+```
+
+The listings are represented as point objects, and what we are interested in is
+the spatial distribution of the listing prices across the market. So unlike the
+airport dataset above where we were focusing on the locations of the points,
+here we extend the analysis to consider the locations and an attribute that is
+measured at each of the locations.
+
+Before we analyze the spatial distribution of the listings, we have to convert
+the data type of the listing column as it is currently encoded as a string:
+
+```python
+gdf.price
+```
+
+```python
+gdf['price'] = gdf['price'].apply(lambda x: x.replace('$', '').replace(',', '')).astype(float)
+```
+
+```python
+gdf.price
+```
+
+
+
+
+```python
+crs = {'init': 'epsg:4326'}
+gdf.crs = crs
+```
+
+
+With the listing variable converted, we can now proceed to visualize the
+spatial distribution:
+
+```python
+gdf.plot(column='price', legend=True, scheme='quantiles', figsize=(16,16))
+```
+
+
+This is somewhat informative, but there are several issues with approaching the
+analysis in this fashion. Chief among these is the occlusion of listings in
+areas with high density. The size of the points also makes it challenging to
+develop a systematic view of the spatial variation in the listing prices.
+
+We can address these issues through spatial aggregation. That is, similar to
+what we did for the airport dataset, we can use a spatial join to associate each
+listing with its enclosing  polygon. Once we have that information, we can use
+database methods to perform aggregation queries to get the average listing
+price per polygon.
+
+
+For our polygons we will use census tracts that we have downloaded for the
+state of California:
+
+```python
+tracts = '../data/sandiego/tl_2019_06_tract.shp'
```

Comment:
This file is no longer provided. Further, using the `acs_tracts_clean.shp` data results in some screw-ups. 
1. The tracts don't need to be filtered anymore 
2. The CRS seems to be incorrectly set? I can't seem to get the two to agree on position correctly with a cursory investigation.